code/game/objects/items/stacks/sheets/sheet_types.dm

```diff
@@ -253,18 +255,36 @@ GLOBAL_LIST_INIT(metal_recipes, list ( \
 	if(build_on.is_blocked_turf())
 		user.balloon_alert(user, "something is blocking the tile!")
 		return ITEM_INTERACT_BLOCKING
-	if(get_amount() < 2)
-		user.balloon_alert(user, "not enough material!")
+
+	var/frame_path = null
+	var/cost = 2 // Default girder cost
+	var/time = 4 SECONDS //Default girder build time
+	var/obj/item/circuitboard/held_board = locate() in user.held_items
+	if(!isnull(held_board))
+		frame_path = text2path("/obj/structure/frame/[istype(held_board, /obj/item/circuitboard/machine) ? "machine" : "computer"]")
```

Comment:
Please make this into a proper ternary/if else instead of this text2path nightmare